ethers-middleware/src/timelag/mod.rs

```diff
@@ -0,0 +1,468 @@
+        from: T,
+        block: Option<BlockId>,
+    ) -> Result<ethers_core::types::U256, Self::Error> {
+        let block = self.normalize_block_id(block).await?;
+
+        self.inner()
+            .get_transaction_count(from, block)
+            .await
+            .map_err(ethers_providers::FromErr::from)
+    }
+
+    async fn call(
+        &self,
+        tx: &TypedTransaction,
+        block: Option<BlockId>,
+    ) -> Result<ethers_core::types::Bytes, Self::Error> {
+        let block = self.normalize_block_id(block).await?;
+
+        self.inner()
+            .call(tx, block)
+            .await
+            .map_err(ethers_providers::FromErr::from)
+    }
+
+    async fn get_balance<T: Into<ethers_core::types::NameOrAddress> + Send + Sync>(
+        &self,
+        from: T,
+        block: Option<BlockId>,
+    ) -> Result<ethers_core::types::U256, Self::Error> {
+        let block = self.normalize_block_id(block).await?;
+        self.inner()
+            .get_balance(from, block)
+            .await
+            .map_err(ethers_providers::FromErr::from)
+    }
+
+    async fn get_transaction_receipt<T: Send + Sync + Into<ethers_core::types::TxHash>>(
+        &self,
+        transaction_hash: T,
+    ) -> Result<Option<TransactionReceipt>, Self::Error> {
+        let receipt = self
+            .inner()
+            .get_transaction_receipt(transaction_hash)
+            .await
+            .map_err(ethers_providers::FromErr::from)?;
+
+        if let Some(receipt) = receipt {
+            if let Some(number) = receipt.block_number {
+                if number > self.get_block_number().await? {
+                    Ok(None)
+                } else {
+                    Ok(Some(receipt))
+                }
+            } else {
+                Ok(Some(receipt))
+            }
+        } else {
+            Ok(None)
+        }
+    }
+
+    async fn get_code<T: Into<ethers_core::types::NameOrAddress> + Send + Sync>(
+        &self,
+        at: T,
+        block: Option<BlockId>,
+    ) -> Result<ethers_core::types::Bytes, Self::Error> {
+        let block = self.normalize_block_id(block).await?;
+
+        self.inner()
+            .get_code(at, block)
+            .await
+            .map_err(ethers_providers::FromErr::from)
+    }
+
+    async fn get_storage_at<T: Into<ethers_core::types::NameOrAddress> + Send + Sync>(
+        &self,
+        from: T,
+        location: ethers_core::types::TxHash,
+        block: Option<BlockId>,
+    ) -> Result<ethers_core::types::TxHash, Self::Error> {
+        let block = self.normalize_block_id(block).await?;
+        self.inner()
+            .get_storage_at(from, location, block)
+            .await
+            .map_err(ethers_providers::FromErr::from)
+    }
+
+    async fn fill_transaction(
+        &self,
+        tx: &mut TypedTransaction,
+        block: Option<BlockId>,
+    ) -> Result<(), Self::Error> {
+        let block = self.normalize_block_id(block).await?;
```

Comment:
I think you can do `self.fill_transaction` given that it's a function on the trait?

Reply:
`self.inner().fill_transaction`